packages/babel-register/src/cache.js

```diff
@@ -32,27 +38,66 @@ export function save() {
     }
   }
 
-  mkdirpSync(path.dirname(FILENAME));
-  fs.writeFileSync(FILENAME, serialised);
+  try {
+    mkdirpSync(path.dirname(FILENAME));
+    fs.writeFileSync(FILENAME, serialised);
+  } catch (e) {
+    switch (e.code) {
+      case "EACCES":
+      case "EPERM":
+        console.warn(
+          `Babel could not write cache to file: ${FILENAME} 
+due to a permission issue. Cache is disabled.`,
+        );
+        cacheDisabled = true;
+        break;
+      case "EROFS":
+        console.warn(
+          `Babel could not write cache to file: ${FILENAME} 
+because it resides in a readonly filesystem. Cache is disabled.`,
+        );
+        cacheDisabled = true;
+        break;
+      default:
+        throw e;
+    }
+  }
 }
 
 /**
  * Load cache from disk and parse.
  */
 
 export function load() {
-  if (process.env.BABEL_DISABLE_CACHE) return;
+  if (isCacheDisabled()) return;
 
   process.on("exit", save);
   process.nextTick(save);
 
-  if (!fs.existsSync(FILENAME)) return;
+  let cacheContent;
 
   try {
-    data = JSON.parse(fs.readFileSync(FILENAME));
-  } catch (err) {
-    return;
+    cacheContent = fs.readFileSync(FILENAME);
+  } catch (e) {
+    switch (e.code) {
+      case "ENOENT":
+        return;
+      case "EACCES":
+      case "EPERM":
+        console.warn(
+          `Babel could not read cache file: ${FILENAME}
+due to a permission issue. Cache is disabled.`,
+        );
+        cacheDisabled = true;
+        return;
+      default:
+        throw e;
+    }
   }
+
+  try {
+    data = JSON.parse(cacheContent);
+  } catch (err) {}
```

Comment:
Nit:

```suggestion
  } catch {}
```

I'm surprised that eslint doesn't warn about the unused variable.

Reply:
> eslint doesn't warn about the unused variable

Because optional catching binding is a stage-4 proposal?